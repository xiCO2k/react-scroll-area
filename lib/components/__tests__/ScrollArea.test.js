'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ScrollArea = require('../ScrollArea');

var _ScrollArea2 = _interopRequireDefault(_ScrollArea);

var _enzyme = require('enzyme');

var _reactTestRenderer = require('react-test-renderer');

var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import jsdom from 'jsdom'

// const doc = new jsdom.JSDOM('<!doctype html><html><body></body></html>')
// const win = doc.defaultView

// global.document = doc
// global.window = win

// propagateToGlobal(win)

// function propagateToGlobal (window) {
//   for (let key in window) {
//     if (!window.hasOwnProperty(key)) continue
//     if (key in global) continue

//     global[key] = window[key]
//   }
// }

it('renders correctly', function () {
    var comp = (0, _enzyme.shallow)(_react2.default.createElement(
        _ScrollArea2.default,
        null,
        'Analytcs'
    ));

    expect(comp).toMatchSnapshot();
});

// it('should have the same width and height of the parent DOM', () => {
//     expect(true).toEqual(true);
// });

// it('shows the scroll track when mouse enter', () => {
//     expect(true).toEqual(true);
// });

// it('should not show the track when mouse enter when the props.trackHidden is true', () => {
//     expect(true).toEqual(true);
// });

// it('should change the class to handlerHover when the scroll handler is hovered', () => {
//     expect(true).toEqual(true);
// });