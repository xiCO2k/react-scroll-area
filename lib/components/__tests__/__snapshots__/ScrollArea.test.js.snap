// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="ScrollArea__outer___37v-o"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="ScrollArea__overflow___3EGck"
      onScroll={[Function]}
      onWheel={[Function]}
    >
      <div
        className="ScrollArea__inner___34zZx"
        style={
          Object {
            "marginRight": -1,
          }
        }
      >
        Analytcs
      </div>
    </div>
    <div
      className="ScrollArea__track___1Rpps ScrollArea__trackHidden_____l1I"
    >
      <div
        className="ScrollArea__handler___hkHU7"
        style={
          Object {
            "height": 0,
            "top": -0,
          }
        }
      />
    </div>
  </div>,
  "nodes": Array [
    <div
      className="ScrollArea__outer___37v-o"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="ScrollArea__overflow___3EGck"
        onScroll={[Function]}
        onWheel={[Function]}
      >
        <div
          className="ScrollArea__inner___34zZx"
          style={
            Object {
              "marginRight": -1,
            }
          }
        >
          Analytcs
        </div>
      </div>
      <div
        className="ScrollArea__track___1Rpps ScrollArea__trackHidden_____l1I"
      >
        <div
          className="ScrollArea__handler___hkHU7"
          style={
            Object {
              "height": 0,
              "top": -0,
            }
          }
        />
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ScrollArea>
        Analytcs
      </ScrollArea>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ScrollArea {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handlerMargin": 4,
        "minTrackHandlerHeight": 70,
        "props": Object {
          "children": "Analytcs",
        },
        "refs": Object {},
        "scrollTrackVisibleTimeout": false,
        "state": Object {
          "innerHeight": 0,
          "innerMargin": -1,
          "isDragging": false,
          "outerHeight": 0,
          "trackHandlerHeight": 0,
          "trackHidden": undefined,
          "trackVisible": false,
        },
        "unMounted": false,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="ScrollArea__outer___37v-o"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
        >
          <div
            className="ScrollArea__overflow___3EGck"
            onScroll={[Function]}
            onWheel={[Function]}
          >
            <div
              className="ScrollArea__inner___34zZx"
              style={
                Object {
                  "marginRight": -1,
                }
              }
            >
              Analytcs
            </div>
          </div>
          <div
            className="ScrollArea__track___1Rpps ScrollArea__trackHidden_____l1I"
          >
            <div
              className="ScrollArea__handler___hkHU7"
              style={
                Object {
                  "height": 0,
                  "top": -0,
                }
              }
            />
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="ScrollArea__outer___37v-o"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
        >
          <div
            className="ScrollArea__overflow___3EGck"
            onScroll={[Function]}
            onWheel={[Function]}
          >
            <div
              className="ScrollArea__inner___34zZx"
              style={
                Object {
                  "marginRight": -1,
                }
              }
            >
              Analytcs
            </div>
          </div>
          <div
            className="ScrollArea__track___1Rpps ScrollArea__trackHidden_____l1I"
          >
            <div
              className="ScrollArea__handler___hkHU7"
              style={
                Object {
                  "height": 0,
                  "top": -0,
                }
              }
            />
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ScrollArea>
    Analytcs
  </ScrollArea>,
}
`;
